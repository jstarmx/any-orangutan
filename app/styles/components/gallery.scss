.gallery {
  display: grid;
  grid-template-columns: repeat(1, 1fr);

  @include min-width(s) {
    grid-template-columns: repeat(2, 1fr);
  }

  @include min-width(l) {
    grid-template-columns: repeat(3, 1fr);
  }

  @include min-width(xl) {
    grid-template-columns: repeat(4, 1fr);
  }

  @include min-width(xxl) {
    grid-template-columns: repeat(5, 1fr);
  }
}
