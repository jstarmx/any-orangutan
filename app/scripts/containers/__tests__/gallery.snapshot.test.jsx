import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { Gallery } from '../gallery';

const store = configureMockStore([thunk])();
const props = {
  activeFilter: 'all',
  favourites: [],
  fetchImages: () => {},
  items: [
    {
      date_taken: '2017-06-03T14:44:43-08:00',
      link: 'http://www.flickr.com/photos/thevixen/34702706110/',
      media: {
        m: 'http://farm5.staticflickr.com/4266/34702706110_e722ea7bf8_m.jpg',
      },
      title: 'Make Them Go Away',
    },
    {
      date_taken: '2017-05-28T11:41:42-08:00',
      link: 'http://www.flickr.com/photos/goonieman70/34925399121/',
      media: {
        m: 'http://farm5.staticflickr.com/4271/34925399121_1a817f3f1c_m.jpg',
      },
      title: 'Monkeying around.',
    },
  ],
};

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={ store }>
      <Gallery { ...props } />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
