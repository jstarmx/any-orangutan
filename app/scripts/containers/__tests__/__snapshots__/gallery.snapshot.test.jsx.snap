// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="gallery"
>
  <div
    className="item "
  >
    <a
      className="item__link"
      href="http://www.flickr.com/photos/thevixen/34702706110/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Make Them Go Away"
        className="item__image"
        src="http://farm5.staticflickr.com/4266/34702706110_e722ea7bf8_m.jpg"
      />
      <div
        className="item__open"
      >
        View on Flickr...
      </div>
    </a>
    <p
      className="item__title"
    >
      Make Them Go Away
    </p>
    <p
      className="item__date"
    >
      3rd Jun 2017
    </p>
    <button
      aria-pressed={false}
      className="item__button"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Add to favourites"
        className="item__icon"
        src="heart_empty.svg"
      />
      favourite
    </button>
  </div>
  <div
    className="item "
  >
    <a
      className="item__link"
      href="http://www.flickr.com/photos/goonieman70/34925399121/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Monkeying around."
        className="item__image"
        src="http://farm5.staticflickr.com/4271/34925399121_1a817f3f1c_m.jpg"
      />
      <div
        className="item__open"
      >
        View on Flickr...
      </div>
    </a>
    <p
      className="item__title"
    >
      Monkeying around.
    </p>
    <p
      className="item__date"
    >
      28th May 2017
    </p>
    <button
      aria-pressed={false}
      className="item__button"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Add to favourites"
        className="item__icon"
        src="heart_empty.svg"
      />
      favourite
    </button>
  </div>
</section>
`;
